packages/babel-plugin/__tests__/stylex-transform-create-test.js

```diff
@@ -36,6 +36,26 @@ function transform(source, opts = {}) {
   }).code;
 }
 
+function transformTypescript(source, opts = {}) {
+  return transformSync(source, {
+    filename: opts.filename,
+    parserOpts: {
+      plugins: ['typescript'],
+    },
+    babelrc: false,
+    plugins: [
+      [
+        stylexPlugin,
+        {
+          runtimeInjection: true,
+          unstable_moduleResolution: { type: 'haste' },
+          ...opts,
+        },
+      ],
+    ],
+  }).code;
+}
+
```

Comment:
Can we update the existing `transform` function to use typescript based on `opts.filename` instead?

That way, we won't need two mostly duplicate functions.

Reply:
BTW, I think that's not enough ( Maybe shouldn't judge whether it is a `tsBlock`) The final conversion result of each compiler

```js

import stylex from "@stylexjs/stylex";
export var X;
((X2) => {
  X2.styles = stylex.create({
    t: {
      color: "red"
    }
  });
})(X || (X = {}));

```

If we want to support namespace, then we should also support this.

Reply:
If this is the output then this is not yet going to be feasible.

We can *hoist* up the generated objects to the top level of the module and replace the original `stylex.create` with a reference to that object. That way we can allow defining `stylex.create` anywhere you want.